### PR TITLE
CIVDT-569: Confirm underpayment details for selected duty type

### PR DESCRIPTION
**Pull request by dannybassi**

### CODE REVIEW

File: app/controllers/DefermentController.scala [hunk 1]
```diff
@@ -92,12 +92,14 @@ class DefermentController @Inject()(identify: IdentifierAction,
...
   }
 
+  // TODO - needs to change once the feature swtich is taken out
   private[controllers] def getHeaderMessage(userAnswers: UserAnswers): String = {
     flowService.dutyType(userAnswers) match {
         case "vat" => "deferment.headingOnlyVAT"
         case "duty" => "deferment.headingDutyOnly"
         case "both" => "deferment.headingVATandDuty"
-      }
+        case _ => "deferment.headingDutyOnly"
```
Comment (ian-ralphs) [inline app/controllers/DefermentController.scala:101]: How is this related to your change? Could you remove it please.

File: app/views/underpayments/UnderpaymentDetailSummaryView.scala.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        h1: components.h1,
+        formHelper: FormWithCSRF,
+        button: components.button
+)
+@(underpaymentType: String, underpaymentSummary: SummaryList)(implicit request: Request[_], messages: Messages)
+
+@getMessage(isTitle: Boolean = true) = @{
+    val key = if(isTitle) "pageTitle" else "pageHeader"
+    val emptySpace = " "
+    val middleMessage = if(underpaymentType == "B00" | underpaymentType == "E00") {
```
Comment (ian-ralphs) [inline app/views/underpayments/UnderpaymentDetailSummaryView.scala.html:28]: confirm the requirement for changing the case of the underpayment type as it is not consistent across the underpayments.
Once this is done create the complete title in the messages file and reference it with:
underpaymentSummaryDetails.A00.pageTitle (or heading)

File: app/views/underpayments/UnderpaymentDetailSummaryView.scala.html [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        messages(s"underpaymentDetails.$underpaymentType.$key.lowercase")
+    } else {
+        messages(s"underpaymentDetails.$underpaymentType.$key")
+    }
+    messages(s"underpaymentDetailsSummary.$key") + emptySpace + middleMessage + emptySpace + messages("underpaymentDetails.dutyMessageEnding")
+}
+
+@layout(
+    pageTitle = getMessage(), customBackLinkUrl = Some(controllers.underpayments.routes.UnderpaymentDetailsController.onLoad(underpaymentType))) {
```
Comment (ian-ralphs) [inline app/views/underpayments/UnderpaymentDetailSummaryView.scala.html:37]: can you pass in the backLink please as this is likely to be dynamic in future pages.

File: app/views/underpayments/UnderpaymentDetailsView.scala.html [hunk 4]
```diff
@@ -25,13 +25,13 @@
 @(form: Form[_], underpaymentType: String, backLink: Call)(implicit request: Request[_], messages: Messages)
 
 @layout(
-    pageTitle = messages(s"underpaymentDetails.$underpaymentType.pageTitle"),
+    pageTitle = messages(s"underpaymentDetails.$underpaymentType.pageTitle") + " " + messages("underpaymentDetails.dutyMessageEnding"),
```
Comment (ian-ralphs) [inline app/views/underpayments/UnderpaymentDetailsView.scala.html:28]: You could just add the space to the dutyEnding message

File: app/controllers/underpayments/UnderpaymentDetailSummaryController.scala [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist._
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.ViewUtils.displayMoney
+import views.html.underpayments.UnderpaymentDetailSummaryView
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class UnderpaymentDetailSummaryController @Inject()(identify: IdentifierAction,
```
Comment (ian-ralphs) [inline app/controllers/underpayments/UnderpaymentDetailSummaryController.scala:36]: I think this might be better renamed as UnderpaymentDetailsConfirmController as this is the point they are confirming the details they have just put in

File: test/assets/messages/underpayments/UnderpaymentDetailSummaryMessages.scala [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    "A10" -> ExpectedContent(
+      beginningMessage + emptySpace + A10pageTitle + emptySpace + endingMessage,
+      beginningMessage + emptySpace + A10pageHeader + emptySpace + endingMessage,
+      Some(A10pageTitle + emptySpace + dueToHMRC)
+    ),
+    "D10" -> ExpectedContent(
+      beginningMessage + emptySpace + D10pageTitle + emptySpace + endingMessage,
+      beginningMessage + emptySpace + D10pageHeader + emptySpace + endingMessage,
+      Some(D10pageTitle + emptySpace + dueToHMRC)
+    ),
```
Comment (ian-ralphs) [inline test/assets/messages/underpayments/UnderpaymentDetailSummaryMessages.scala:103]: one too many commas

File: app/controllers/underpayments/UnderpaymentDetailSummaryController.scala [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+                                                   )
+  extends FrontendController(mcc) with I18nSupport {
+
+  def onLoad(underpaymentType: String): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    val underpaymentDetail = request.userAnswers.get(UnderpaymentDetailsPage).getOrElse(UnderpaymentAmount(0, 0))
+    Future.successful(Ok(view(underpaymentType, summaryList(underpaymentType, underpaymentDetail))))
+  }
+
+  def onSubmit(underpaymentType: String): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    val currentUnderpaymentDetail = request.userAnswers.get(UnderpaymentDetailsPage).getOrElse(UnderpaymentAmount(0, 0))
```
Comment (ian-ralphs) [inline app/controllers/underpayments/UnderpaymentDetailSummaryController.scala:51]: I think we need to look at how we are handling this getOrElse case because doing it here could potentially save 0, 0 in the underpayment when they haven't confirmed that value.

**Review by ian-ralphs — CHANGES_REQUESTED** [review]

File: app/views/underpayments/UnderpaymentDetailsView.scala.html (hunk 4 above)
Comment (tvaidotas) [reply app/views/underpayments/UnderpaymentDetailsView.scala.html:28]: pretty sure we tried it and it gets cleaned as a spare whitespace

File: app/views/underpayments/UnderpaymentDetailSummaryView.scala.html (hunk 3 above)
Comment (tvaidotas) [reply app/views/underpayments/UnderpaymentDetailSummaryView.scala.html:37]: more efficient for what we need currently to have it like this

File: app/controllers/DefermentController.scala (hunk 1 above)
Comment (tvaidotas) [reply app/controllers/DefermentController.scala:101]: this is a temporary fix to have deferment page working when using the new flow to make a submission, in the next story this will need to be reworked
